nbgrader_to_canvas/upload_grades.py

```diff
@@ -0,0 +1,379 @@
+                    app.logger.error('Error deleting comments: {}'.format(ex))
+                if counter < self._num_students:
+                    counter += 1
+                    update_status(self._form_nb_assign_name, self._course_id, completion=5+15*counter/self._num_students)
+            
+
+    def _get_feedbacks(self):
+        update_status(self._form_nb_assign_name, self._course_id, completion=20, status='Fetching Feedback')
+        counter = 0
+        with Gradebook("sqlite:////mnt/nbgrader/"+self._nbgrader_course+"/grader/gradebook.db") as gb:
+            for student in gb.students:
+                if not os.path.isdir('/mnt/nbgrader/'+self._nbgrader_course+'/grader/feedback/'+student.id+'/'+self._form_nb_assign_name):
+                    continue
+                for file in os.listdir('/mnt/nbgrader/'+self._nbgrader_course+'/grader/feedback/'+student.id+'/'+self._form_nb_assign_name):
+                    if file.endswith('.html'):
+                        files = {'file': open('/mnt/nbgrader/'+self._nbgrader_course+'/grader/feedback/'+student.id+'/'+self._form_nb_assign_name+'/'+file, 'rb')}
+                        r_json = self._prepare_feedback(student, file)  
+                        response = self._upload_feedback(r_json, files)
+                        r_json = response.json()
+                        if response.status_code >= 300 and response.status_code < 400:
+                            r_json = self._confirm_feedback(r_json) 
+                        self.nbgrader_feedback[student.id] = r_json['id']
+                if counter < self._num_students:
+                    counter += 1
+                    update_status(self._form_nb_assign_name, self._course_id, completion=20+15*counter/self._num_students)
+
+    def _prepare_feedback(self, student, file):
+        auth_header = {'Authorization': 'Bearer ' + self._flask_session['api_key']}
+        upload_args = {'name':file}
+        response = requests.post(
+            "{}api/v1/courses/{}/assignments/{}/submissions/{}/comments/files".format(
+                settings.API_URL, self._course_id, self.assignment_to_upload.id, self.canvas_students[student.id]
+            ),
+            headers = auth_header,
+            data = upload_args
+        )
+        return response.json()
+    
+    def _upload_feedback(self, r_json, files):
+        response = requests.post(
+                r_json['upload_url'],
+                allow_redirects=False,
+                data=r_json['upload_params'],
+                files= files
+            )
+        return response        
+    
+    def _confirm_feedback(self, r_json):
+        headers = {'Content-Length':'0', 'Authorization': 'Bearer ' + self._flask_session['api_key']}
+        response = requests.get(
+            r_json['location'],
+            headers=headers
+        )
+        return response
+
+    # Updates grades for given assignment. Returns progress resulting from upload attempt.
```

Comment:
would it be useful to describe what "progress" means in this comment (i.e., canvas assignment submission progress object information)